core/src/main/java/org/bitcoinj/core/BitcoinSerializer.java

```diff
@@ -378,4 +404,19 @@ public BitcoinPacketHeader(ByteBuffer in) throws ProtocolException, BufferUnderf
             cursor += 4;
         }
     }
+
```

Comment:
I would say adding equals/hashCode to BitcoinSerializer could go into a separate commit?

Reply:
I added a variant of this via 3cbb10334edae29b86dbc08686bf61c487851061.